### PR TITLE
docs: update Docker instructions

### LINKED ISSUE #876: Unable to run make command inside docker
Latest installed ORFS using docker mode.
I ran following commands:
```
export OS_NAME=centos7
docker run -it -u $(id -u ${USER}):$(id -g ${USER}) openroad/flow-$OS_NAME-builder
source setup_env.sh
cd flow
make
```
Failed with below message:
```
[INFO][FLOW] Using platform directory ./platforms/nangate45
mkdir: cannot create directory './objects': Permission denied
make: *** [objects/nangate45/gcd/base/lib/NangateOpenCellLibrary_typical.lib] Error 1
bash-4.2$ ls
```
How to resolve this permission issue?
Following instruction from [here](https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithDocker.html#verify-installation)

_Originally posted by @vijayank88 in https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/discussions/862_

### PR DESCRIPTION
**Pull request by vvbandeira**

### CODE REVIEW

File: docs/user/BuildWithDocker.md [hunk 1]
```diff
@@ -70,3 +55,19 @@ cd flow
 make
 exit
 ```
+
+## Build Docker Image for Different OS
+
+The following instructions build the docker image with a parameterized OS
+in two stages. These are for CI and developers that wish to use an OS other
+than CentOS 7, regular users should use the steps from previous sections.
```
Comment (rovinski) [inline docs/user/BuildWithDocker.md]:
```suggestion
than CentOS 7; regular users should use the steps from previous sections.
```

File: docs/user/BuildWithDocker.md [hunk 2]
```diff
@@ -70,3 +55,19 @@ cd flow
...
+Dev stage for installing all the dependencies and packages that are needed
+to run OpenROAD and OpenROAD Flow Scripts. The build stage actually builds
+all of the binaries needed to run flow (i.e., `openroad` and `yosys`).
```
Comment (rovinski) [inline docs/user/BuildWithDocker.md]:
```suggestion
The dev stage installs all dependencies and packages required to run OpenROAD
and OpenROAD Flow Scripts. The build stage generates all binaries needed to run
the flow (i.e., `openroad` and `yosys`).
```

**Comment by vijayank88** [comment]
@vvbandeira 
I am able run gcd design successfully with updated docker command. 
But I'm not able to view `openroad -gui` inside docker.
```
bash-4.2$ openroad -gui
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
```
Is anything additional to append with docker command to get display access?

**Comment by vvbandeira** [comment]
> Is anything additional to append with docker command to get display access?

See https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/901/commits/1f7cadd6fb4fb6bd76852348d78e1c52477fa85b

**Comment by vijayank88** [comment]
@vvbandeira 
Both the commands working as expected. LGTM!!!

File: docs/user/BuildWithDocker.md [hunk 3]
```diff
@@ -70,3 +55,35 @@ cd flow
 make
 exit
 ```
+
+## Enable GUI support
+
+To use the GUI feature you will need to start the docker with the following command:
+
+```
+docker run --rm -it \
```
Comment (maliberty) [inline docs/user/BuildWithDocker.md:64]: It is good to have this documented.  You could add a flow/Makefile target like 'make mount' to do this more easily.

Comment (vvbandeira) [reply docs/user/BuildWithDocker.md:64]: Merging docs now to deploy new readthedocs page. Will add the command later.

**Pull request merged**